pvlib/pvsystem.py

```diff
@@ -1372,6 +1398,67 @@ def get_iam(self, aoi, iam_model='physical'):
         else:
             raise ValueError(model + ' is not a valid IAM model')
 
+    def dc_ohms_from_percent(self):
+        """
+        Calculates the equivalent resistance of the wires using
+        :py:func:`pvlib.pvsystem.dc_ohms_from_percent`
+
+        Makes use of array module parameters according to the
+        following DC models:
+        CEC:
+            `self.module_parameters["V_mp_ref"]`,
+            `self.module_parameters["I_mp_ref"]`,
+        SAPM:
+            `self.module_parameters["Vmpo"]`,
+            `self.module_parameters["Impo"]`,
+        PVsyst-like or other:
+            `self.module_parameters["Vmpp"]`,
+            `self.module_parameters["Impp"]`,
+
+        Other array parameters that are used are:
+        `self.losses_parameters["dc_ohmic_percent"]`,
+        `self.modules_per_string`, and
+        `self.strings`.
+
+        See :py:func:`pvlib.pvsystem.dc_ohms_from_percent` for more details.
+        """
+
+        kwargs = _build_kwargs(['dc_ohmic_percent'],
+                               self.array_losses_parameters)
+
+        # add relevent Vmp and Imp parameters from CEC parameters
+        if all([elem in self.module_parameters
+                for elem in ['V_mp_ref', 'I_mp_ref']]):
+            kwargs.update({'vmp_ref': self.module_parameters['V_mp_ref'],
+                           'imp_ref': self.module_parameters['I_mp_ref']})
+
+        # add relevant Vmp and Imp parameters from SAPM parameters
+        elif all([elem in self.module_parameters
+                  for elem in ['Vmpo', 'Impo']]):
+            kwargs.update({'vmp_ref': self.module_parameters['Vmpo'],
+                           'imp_ref': self.module_parameters['Impo']})
+
+        # add relevant Vmp and Imp parameters if they are PVsyst-like
+        elif all([elem in self.module_parameters
+                  for elem in ['Vmpp', 'Impp']]):
+            kwargs.update({'vmp_ref': self.module_parameters['Vmpp'],
+                           'imp_ref': self.module_parameters['Impp']})
+
+        # raise error if relevant Vmp and Imp parameters are not found
+        else:
+            raise ValueError('Parameters for Vmp and Imp could not be found '
+                             'in the array module parameters. Module '
+                             'parameters must include one set of '
+                             '{"V_mp_ref", "I_mp_Ref"}, '
+                             '{"Vmpo", "Impo"}, or '
+                             '{"Vmpp", "Impp"}.'
+                             )
+
+        kwargs.update({'modules_per_string': self.modules_per_string,
+                       'strings': self.strings})
```

Comment:
continuing the kwargs theme... this would become `kwargs = {'modules_per_string'...`